OWML.ModHelper/ModInputHandler.cs

```diff
@@ -0,0 +1,447 @@
+					neg = (KeyCode)(typeof(SingleAxisCommand).GetField("_gamepadKeyCodeNegative", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+				}
+				else
+				{
+					pos = (KeyCode)(typeof(SingleAxisCommand).GetField("_keyPositive", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					neg = (KeyCode)(typeof(SingleAxisCommand).GetField("_keyNegative", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+				}
+				if (Input.GetKey(pos) && ModInputHandler._self.ShouldIgnore(pos))
+				{
+					curval -= 1f * axisdir;
+					//DebugInput.logger.Log("succesfully ignored " + pos.ToString());
+				}
+				if (Input.GetKey(neg) && ModInputHandler._self.ShouldIgnore(neg))
+				{
+					curval += 1f * axisdir;
+					//DebugInput.logger.Log("succesfully ignored " + neg.ToString());
+				}
+				val.SetValue(__instance, curval);
+			}
+
+			public static void DAxisUpdPost(DoubleAxisCommand __instance)
+			{
+				ModInputHandler._self.RegisterGamesBinding(__instance);
+				if (!OWInput.UsingGamepad())
+				{
+					KeyCode pos, neg;
+					FieldInfo val = typeof(DoubleAxisCommand).GetField("_value", BindingFlags.NonPublic | BindingFlags.Instance);
+					Vector2 curval = (Vector2)(val.GetValue(__instance));
+					pos = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardXPos", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					neg = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardXNeg", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					if (Input.GetKey(pos) && ModInputHandler._self.ShouldIgnore(pos))
+					{
+						curval.x -= 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + pos.ToString());
+					}
+					if (Input.GetKey(neg) && ModInputHandler._self.ShouldIgnore(neg))
+					{
+						curval.x += 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + neg.ToString());
+					}
+					pos = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardYPos", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					neg = (KeyCode)(typeof(DoubleAxisCommand).GetField("_keyboardYNeg", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
+					if (Input.GetKey(pos) && ModInputHandler._self.ShouldIgnore(pos))
+					{
+						curval.y -= 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + pos.ToString());
+					}
+					if (Input.GetKey(neg) && ModInputHandler._self.ShouldIgnore(neg))
+					{
+						curval.y += 1f;
+						//				DebugInput.console.WriteLine("succesfully ignored " + neg.ToString());
+					}
+					val.SetValue(__instance, curval);
+				}
+			}
+		}
+
+		public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper patcher)
+		{
+			_console = console;
+			_logger = logger;
+			patcher.AddPrefix<SingleAxisCommand>("ClearBinding", typeof(InputInterceptor), "SAxisRemPre");
```

Comment:
```suggestion
			patcher.AddPrefix<SingleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.SAxisRemPre));
```

(same for other patches)

Reply:
done